
This patch adds PyGame package to buildroot.
Currently added to "graphic libraries".

Movie and Scrap modules are NOT tested: movie requires spmeg (SDL MPEG library)
and Scrap needs X11.
numeric*array modules requires Numeric Python package which is not in BR yet.

Inspired by a patch from Jean Schurger and the work of the OpenEmbedded team.

Signed-off-by: Julien Boibessot <julien.boibessot@armadeus.com>

Index: buildroot/package/pygame/Config.in
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/pygame/Config.in	2010-06-23 14:11:36.000000000 +0200
@@ -0,0 +1,92 @@
+config BR2_PACKAGE_PYGAME
+	bool "pygame"
+	depends on BR2_PACKAGE_PYTHON
+	select BR2_PACKAGE_SDL
+	default n
+	help
+	  Pygame is a set of Python modules designed for writing games.
+	  Pygame adds functionality on top of the excellent SDL library.
+
+	  http://www.pygame.org/
+
+menu "pygame optional modules selection"
+	depends on BR2_PACKAGE_PYGAME
+
+config BR2_PACKAGE_PYGAME_IMAGE
+	bool "pygame.image"
+	default y
+	select BR2_PACKAGE_SDL_IMAGE
+	help
+	  pygame module for image transfer
+
+menu "Image formats"
+	depends on BR2_PACKAGE_PYGAME_IMAGE
+
+config BR2_PACKAGE_PYGAME_IMAGE_PNG
+	bool "png"
+	default y
+	select BR2_PACKAGE_SDL_IMAGE_PNG
+
+config BR2_PACKAGE_PYGAME_IMAGE_JPEG
+	bool "jpeg"
+	default n
+	select BR2_PACKAGE_SDL_IMAGE_JPEG
+
+endmenu
+
+config BR2_PACKAGE_PYGAME_FONT
+	bool "pygame.font"
+	default n
+	select BR2_PACKAGE_SDL_TTF
+	help
+	  pygame module for loading and rendering fonts
+
+config BR2_PACKAGE_PYGAME_MIXER
+	bool "pygame.mixer"
+	default n
+	select BR2_PACKAGE_SDL_MIXER
+	help
+	  pygame module for loading and playing sounds
+
+config BR2_PACKAGE_PYGAME_MIXER_MUSIC
+	bool "pygame.mixer.music"
+	default n
+	depends on BR2_PACKAGE_PYGAME_MIXER
+	help
+	  pygame module for controlling streamed audio
+
+config BR2_PACKAGE_PYGAME_SNDARRAY
+	bool "pygame.sndarray"
+	default n
+	depends on BR2_PACKAGE_PYGAME_MIXER && BR2_PACKAGE_PYTHON_NUMERIC
+	help
+	  pygame module for accessing sound sample data
+
+comment "pygame.sndarray depends on Python's Numeric extension that is not (yet) supported"
+	depends on !BR2_PACKAGE_PYGAME_SNDARRAY
+
+config BR2_PACKAGE_PYGAME_SURFARRAY
+	bool "pygame.surfarray"
+	default n
+	depends on BR2_PACKAGE_PYTHON_NUMERIC
+	help
+	  pygame module for accessing surface pixel data using array interfaces
+
+comment "pygame.surfarray depends on Python's Numeric extension that is not (yet) supported"
+	depends on !BR2_PACKAGE_PYGAME_SURFARRAY
+
+config BR2_PACKAGE_PYGAME_MOVIE
+	bool "pygame.movie"
+	default n
+	select BR2_PACKAGE_SMPEG
+	help
+	  pygame module for playback of mpeg video
+
+config BR2_PACKAGE_PYGAME_SCRAP
+	bool "pygame.scrap"
+	default n
+	depends on BR2_PACKAGE_SDL_X11
+	help
+	  pygame module for clipboard support (X11 needed)
+
+endmenu
Index: buildroot/package/pygame/pygame.mk
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/pygame/pygame.mk	2010-06-23 14:11:36.000000000 +0200
@@ -0,0 +1,100 @@
+#
+# Pygame
+#
+
+PYGAME_VERSION:= 1.8.1release
+PYGAME_SOURCE:= pygame-$(PYGAME_VERSION).tar.gz
+PYGAME_SITE:= http://www.pygame.org/ftp/
+PYGAME_DIR:= $(BUILD_DIR)/pygame-$(PYGAME_VERSION)
+PYGAME_BINARY:= pygame
+PYGAME_CAT:= $(ZCAT)
+
+
+ifeq ($(BR2_PACKAGE_PYGAME_IMAGE),y)
+PYGAME_OPT_DEPENDS += sdl_image
+endif
+ifeq ($(BR2_PACKAGE_PYGAME_FONT),y)
+PYGAME_OPT_DEPENDS += sdl_ttf
+endif
+ifeq ($(BR2_PACKAGE_PYGAME_MIXER),y)
+PYGAME_OPT_DEPENDS += sdl_mixer
+endif
+ifeq ($(BR2_PACKAGE_PYGAME_MOVIE),y)
+PYGAME_OPT_DEPENDS += smpeg
+endif
+ifeq ($(BR2_PACKAGE_PYGAME_SCRAP),y)
+PYGAME_OPT_DEPENDS += ?X11?
+endif
+PYGAME_SDL_CFLAGS=$(shell $(STAGING_DIR)/usr/bin/sdl-config --cflags)
+
+
+$(DL_DIR)/$(PYGAME_SOURCE):
+	$(WGET) -P $(DL_DIR) $(PYGAME_SITE)/$(PYGAME_SOURCE)
+
+pygame-source: $(DL_DIR)/$(PYGAME_SOURCE)
+
+$(PYGAME_DIR)/.unpacked: $(DL_DIR)/$(PYGAME_SOURCE)
+	$(PYGAME_CAT) $(DL_DIR)/$(PYGAME_SOURCE) | tar -C $(BUILD_DIR) $(TAR_OPTIONS) -
+	touch $@
+
+$(PYGAME_DIR)/.configured: $(PYGAME_DIR)/.unpacked
+	cp -f package/pygame/Setup $(PYGAME_DIR)
+	echo "$(PYGAME_SDL_INCLUDES)"
+	$(SED) 's#SDL_CUSTOM_CFLAGS_REPLACE_ME#$(PYGAME_SDL_CFLAGS)#' $(PYGAME_DIR)/Setup
+ifeq ($(BR2_PACKAGE_SDL_X11),y)
+	sed -i "s/-DDISABLE_X11//" $(PYGAME_DIR)/Setup
+endif
+ifeq ($(BR2_PACKAGE_PYGAME_IMAGE),y)
+	sed -i "s/#image/image/" $(PYGAME_DIR)/Setup
+endif
+ifeq ($(BR2_PACKAGE_PYGAME_FONT),y)
+	sed -i "s/#font/font/" $(PYGAME_DIR)/Setup
+endif
+ifeq ($(BR2_PACKAGE_PYGAME_MIXER),y)
+	sed -i "s/#mixer/mixer/" $(PYGAME_DIR)/Setup
+endif
+ifeq ($(BR2_PACKAGE_PYGAME_SNDARRAY),y)
+	sed -i "s/#_numericsndarray/_numericsndarray/" $(PYGAME_DIR)/Setup
+endif
+ifeq ($(BR2_PACKAGE_PYGAME_SURFARRAY),y)
+	sed -i "s/#_numericsurfarray/_numericsurfarray/" $(PYGAME_DIR)/Setup
+endif
+ifeq ($(BR2_PACKAGE_PYGAME_MOVIE),y)
+	sed -i "s/#movie/movie/" $(PYGAME_DIR)/Setup
+endif
+ifeq ($(BR2_PACKAGE_PYGAME_SCRAP),y)
+	sed -i "s/#scrap/scrap/" $(PYGAME_DIR)/Setup
+endif
+
+$(PYGAME_DIR)/.built: $(PYGAME_DIR)/.configured
+	(cd $(PYGAME_DIR); \
+		$(TARGET_CONFIGURE_OPTS) \
+		CFLAGS="$(TARGET_CFLAGS)" \
+		LDSHARED="$(TARGET_CROSS)gcc -shared" \
+		$(PYTHON_DIR)/hostpython setup.py build );
+	touch $@
+
+$(PYGAME_DIR)/.installed: $(PYGAME_DIR)/.built
+	(cd $(PYGAME_DIR); \
+		$(TARGET_CONFIGURE_OPTS) \
+		CFLAGS="$(TARGET_CFLAGS)" \
+		LDSHARED="$(TARGET_CROSS)gcc -shared" \
+		$(PYTHON_DIR)/hostpython setup.py install --prefix=$(TARGET_DIR)/usr);
+	touch $@
+
+pygame: python sdl $(PYGAME_OPT_DEPENDS) $(PYGAME_DIR)/.installed
+
+pygame-clean:
+	rm -rf $(TARGET_DIR)/usr/lib/python2.4/pygame/site-packages/
+
+pygame-dirclean:
+	rm -rf $(PYGAME_DIR)
+
+#############################################################
+#
+# Toplevel Makefile options
+#
+#############################################################
+ifeq ($(strip $(BR2_PACKAGE_PYGAME)),y)
+TARGETS+=pygame
+endif
Index: buildroot/package/Config.in
===================================================================
--- buildroot.orig/package/Config.in	2010-06-23 14:11:31.000000000 +0200
+++ buildroot/package/Config.in	2010-06-23 14:11:36.000000000 +0200
@@ -104,6 +104,7 @@
 source "package/imagemagick/Config.in"
 source "package/linux-fusion/Config.in"
 source "package/lite/Config.in"
+source "package/pygame/Config.in"
 source "package/sawman/Config.in"
 source "package/sdl/Config.in"
 source "package/sdl_image/Config.in"
Index: buildroot/package/pygame/Setup
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/pygame/Setup	2010-06-23 14:11:36.000000000 +0200
@@ -0,0 +1,64 @@
+#This Setup file is used by the setup.py script to configure the
+#python extensions. You will likely use the "config.py" which will
+#build a correct Setup file for you based on your system settings.
+#If not, the format is simple enough to edit by hand. First change
+#the needed commandline flags for each dependency, then comment out
+#any unavailable optional modules in the first optional section.
+
+
+SDL = -lSDL -DDISABLE_X11 SDL_CUSTOM_CFLAGS_REPLACE_ME
+FONT = -lSDL_ttf
+IMAGE = -lSDL_image
+MIXER = -lSDL_mixer
+SMPEG = -D_REENTRANT -lsmpeg -lSDL
+PNG = -lpng
+JPEG = -ljpeg
+SCRAP = -lX11
+
+#DEBUG = -C-W -C-Wall
+DEBUG =
+
+#the following modules are optional. you will want to compile
+#everything you can, but you can ignore ones you don't have
+#dependencies for, just comment them out
+
+#imageext src/imageext.c $(SDL) $(IMAGE) $(PNG) $(JPEG) $(DEBUG)
+#font src/font.c $(SDL) $(FONT) $(DEBUG)
+#mixer src/mixer.c $(SDL) $(MIXER) $(DEBUG)
+#mixer_music src/music.c $(SDL) $(MIXER) $(DEBUG)
+#_numericsurfarray src/_numericsurfarray.c $(SDL) $(DEBUG)
+#_numericsndarray src/_numericsndarray.c $(SDL) $(MIXER) $(DEBUG)
+#movie src/movie.c $(SDL) $(SMPEG) $(DEBUG)
+#scrap src/scrap.c $(SDL) $(SCRAP) $(DEBUG)
+
+#experimental new movie movie. requires libavcodec and libavformat.
+#add any necessary compile flags to this line and uncomment.
+#movieext src/movie.c  src/ffmovie.c $(SDL) -lavcodec -lavformat
+
+
+#these modules are required for pygame to run. they only require
+#SDL as a dependency. these should not be altered
+
+base src/base.c $(SDL) $(DEBUG)
+cdrom src/cdrom.c $(SDL) $(DEBUG)
+color src/color.c $(SDL) $(DEBUG)
+constants src/constants.c $(SDL) $(DEBUG)
+display src/display.c $(SDL) $(DEBUG)
+event src/event.c $(SDL) $(DEBUG)
+fastevent src/fastevent.c src/fastevents.c $(SDL) $(DEBUG)
+key src/key.c $(SDL) $(DEBUG)
+mouse src/mouse.c $(SDL) $(DEBUG)
+rect src/rect.c $(SDL) $(DEBUG)
+rwobject src/rwobject.c $(SDL) $(DEBUG)
+surface src/surface.c src/alphablit.c src/surface_fill.c $(SDL) $(DEBUG)
+surflock src/surflock.c $(SDL) $(DEBUG)
+time src/time.c $(SDL) $(DEBUG)
+joystick src/joystick.c $(SDL) $(DEBUG)
+draw src/draw.c $(SDL) $(DEBUG)
+image src/image.c $(SDL) $(DEBUG)
+overlay src/overlay.c $(SDL) $(DEBUG)
+transform src/transform.c src/rotozoom.c src/scale2x.c $(SDL) $(DEBUG)
+mask src/mask.c src/bitmask.c $(SDL) $(DEBUG)
+bufferproxy src/bufferproxy.c $(SDL) $(DEBUG)
+pixelarray src/pixelarray.c $(SDL) $(DEBUG)
+
